Export ClientStatsHandler and ServerStatsHandler from grpcstats

rakyll opened the pull request:
Exporting these handlers makes the package more compatible with
the grpctrace package and allow us to document the behavior via godoc.

/cc @dfawley

jcd2 commented:
Exporting the types looks fine but I don't think we should delete the New... constructors.  Being able to use zero-values directly is nice, but while the constructor is currently trivial, we don't want to rule out adding more to it later.

rakyll commented:
I added the constructors back and added a TODO to remove them if they are not necessary. If the zero value will be always usable (seems like it will be), it is not encouraged to have a constructor for the zero value.
